### PR TITLE
Gruzjan/create game

### PR DESCRIPTION
**Pull request by gnwisniewski**

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -2,33 +2,45 @@ cmake_minimum_required (VERSION 3.11)
 
 add_subdirectory(third_party)
 
-project(smk-starter
+project(lysy-bingo
```
Comment (m1k53r) [inline CMakeLists.txt:5]: does this work for you? I had some problems earlier that it generated lysy-bingo.js but tried to include smk-starter.js in html

File: src/headers/CreateGame.hpp [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+                                                            L"\"Mikisheesh\"", L"\"Wlochaty\"", L"\"Panie technik\"",
+                                                            L"Praca w grupach", L"Uwaga za mundur", L"Pompki",
+                                                            L"\"Lendzionek\"", L"\"Czego?\"", L"\"Co tam brzeczysz?\"",
+                                                            L"\"Dzieci\"", L"Zapowiedzana kartkowka", L"Zapowiedziany sprawdzian",
+                                                            L"Grzybowski incydent", L"Narzekanie na mazak", L"Musi byc C++",
+                                                            L"Zla nieobecnosc", L"*Puszcza film*", L"\"Nie psuj mi ...!\"",
+                                                            L"\"Krzesla!\"", L"Lagodniejsze traktowanie kobiet",
+                                                            L"Muzyka w tle", L"*Slychac klucze*", L"Godzina wychowawcza",
+                                                            L"Problem z pendrivem", L"Absurdalna uwaga", L"Absurdalna pochwala"}}
+        };
```
Comment (m1k53r) [inline src/headers/CreateGame.hpp:35]: prettier will be needed

File: src/scenes/CreateGame.cpp [hunk 3]
```diff
@@ -0,0 +1,235 @@
...
+
+std::string CreateGame::generateCode() {
+  static const char alphabet[] = "ABCDEFGHIJKLMNOPQRSTUVWXYZ";
+  std::string result;
+  result.reserve(6);
+  for(int i = 0; i < 6; i++)
+    result += alphabet[rand() % (sizeof(alphabet) - 1)];
+    
+  return result;
+}
```
Comment (m1k53r) [inline src/scenes/CreateGame.cpp:235]: code gets generated on backend so we dont need this

File: src/scenes/GameMenu.cpp [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+  for (auto& path : path_list) {
+    auto file = std::ifstream(path + "/bingo");
+    if (!file)
+      continue;
+
+    std::string line;
+    if (std::getline(file, line) && line == "bingo") {
+      resource_path = path;
+    }
+  }
```
Comment (m1k53r) [inline src/scenes/GameMenu.cpp:53]: note to myself to remove this

File: src/utils/Tile.cpp [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+
+void Tile::setPassword(std::wstring password){
+    this->password = password;
+}
+
+std::wstring Tile::getPassword(){
+    return password;
+}
+
+void Tile::setCheck(bool check){ //TODO: WTF
```
Comment (m1k53r) [inline src/utils/Tile.cpp:38]: this hits hard

**Review by m1k53r — APPROVED** [review]
looks g just remember to later remove code generator

File: CMakeLists.txt (hunk 1 above)
Comment (gnwisniewski) [reply CMakeLists.txt:5]: Have you tried changing the linked .js file in index.html?
Change
`<script async src="smk-starter.js"></script>`
To
`<script async src="lysy-bingo.js"></script>`

Comment (m1k53r) [reply CMakeLists.txt:5]: Yeah I did that, but I had to change it every time I ran `emcmake cmake ..` on an empty build dir

**Pull request merged**